### PR TITLE
AgeChanged reason added

### LINKED ISSUE #587: Do not reduce age on inactivity
Reducing it to 0 directly is harsh, have fair mech

### LINKED ISSUE #590: Add reason to AgeChange event
- [ ] Reason is required on web

### LINKED ISSUE #599: Remove redundant age change
There is issue with this age mech itself 
How there can be AgeChanged.InactivityPenalty and AgeChanged.MissedEpoch both ?

so at first in givePenalties, we call giveInactivityPenalties which says penalise this much for age 
= no of epcochs missed * 10000 

then we come to this if 
which directly reduce it to 0 

isn't this double ?
what was purpose of this if ?

_Originally posted by @abhishekvispute in https://github.com/razor-network/contracts/pull/591#discussion_r757294685_

### PR DESCRIPTION
**Pull request by SamAg19**
fixes #590 
fixes #599 
fixes #587

### CODE REVIEW

**Review by SkandaBhat — APPROVED** [review]

File: contracts/Core/RewardManager.sol [hunk 1]
```diff
@@ -100,7 +100,7 @@ contract RewardManager is Initializable, Constants, RewardManagerParams, IReward
         Structs.Staker memory thisStaker = stakeManager.getStaker(stakerId);
         uint32 epochLastRevealed = voteManager.getEpochLastRevealed(stakerId);
         if (epochLastRevealed != 0 && epochLastRevealed < epoch - 1) {
-            stakeManager.setStakerAge(epoch, thisStaker.id, 0);
+            stakeManager.setStakerAge(epoch, thisStaker.id, 0, AgeChanged.MissedEpoch);
```
Comment (abhishekvispute) [inline contracts/Core/RewardManager.sol:103]: Hey I think there is issue with this age mech itself 
How there can be AgeChanged.InactivityPenalty and AgeChanged.MissedEpoch both ?

so at first in givePenalties, we call giveInactivityPenalties which says penalise this much for age 
= no of epcochs missed * 10000 

then we come to this if 
which directly reduce it to 0 

isn't this double ?
what was purpose of this if ?

Comment (abhishekvispute) [reply contracts/Core/RewardManager.sol:103]: @razor-network/engineering

**Review by abhishekvispute — CHANGES_REQUESTED** [review]

File: contracts/Core/RewardManager.sol (hunk 1 above)
Comment (abhishekvispute) [reply contracts/Core/RewardManager.sol:103]: looks like this if is not needed, what was its purpose 
anyways we are thinking removing this step where we reduce age directly to 0 
so should we remove this if entirely ? @hrishikeshio @SkandaBhat 

Related Issue https://github.com/razor-network/contracts/issues/587

Comment (hrishikeshio) [reply contracts/Core/RewardManager.sol:103]: yes but the if and return is still needed. else the following calculation will not be correct.

**Review by hrishikeshio — CHANGES_REQUESTED** [review]

**Review by hrishikeshio — APPROVED** [review]

**Review by abhishekvispute — APPROVED** [review]

**Pull request merged**